src/main/java/pl/allegro/tech/discovery/consul/recipes/json/JsonValueReader.java

```diff
@@ -11,10 +11,25 @@ public static <T> T requiredValue(Map serviceProps, String property, Class<T> cl
             throw new JsonDecoder.JsonDecodeException(property + " property is missing in JSON. " +
                     "This may indicate that there are incompatible changes in Consul API.");
         }
+        mustBeAbleToCastToClass(property, clazz, value);
+        return (T) value;
+    }
+
+
+    @SuppressWarnings("unchecked")
+    public static <T> T optionalValue(Map serviceProps, String property, Class<T> clazz) {
```

Comment:
I guess this method should be named more like `nullableValue`, or return Optional (and do not build it in `ServiceInstance` then)

Reply:
renamed as proposed